### PR TITLE
Working with generated files

### PR DESCRIPTION
**Pull request by gsarma**
@rgerkin 
Additions to allow for specifying locations of generated files.  This was motivated by changes to the OpenWorm tests repo.

### CODE REVIEW

File: sciunit/utils.py [hunk 1]
```diff
@@ -162,24 +187,43 @@ def convert_and_execute_notebook(self, name):
...
 
+    def gen_file_path(self, name):
+        """
+        Returns full path to generated files.  Checks to see if directory
+        exists where generated files are stored and creates one otherwise.
+        """
+        relative_path = self.convert_path(name)
+        file_path = self.get_path("%s.ipynb"%relative_path)
+        parent_path = recApply(os.path.dirname, self.genFileLevel)(file_path)
+        genFileName = name if isinstance(name,str) else name[1] #Name of generated file
```
Comment (rgerkin) [inline sciunit/utils.py]: See comment above about camelCase.

Comment (gsarma) [reply sciunit/utils.py]: Fixed in latest commit.

File: sciunit/utils.py [hunk 2]
```diff
@@ -77,11 +86,26 @@ def assert_dimensionless(value):
 class NotebookTools(object):
     """A class for manipulating and executing Jupyter notebooks."""
 
+    path = '' # Relative path to the parent directory of the notebook.
+    genDirName = 'GeneratedFiles' # Name of directory where files generated by do_notebook are stored
```
Comment (rgerkin) [inline sciunit/utils.py]: camelCase -> snake_case

File: sciunit/utils.py [hunk 3]
```diff
@@ -77,11 +86,26 @@ def assert_dimensionless(value):
...
+    genFileLevel = 2 # Number of levels up from notebook directory where generated files are stored
```
Comment (rgerkin) [inline sciunit/utils.py]: See above

File: sciunit/utils.py [hunk 4]
```diff
@@ -162,24 +187,43 @@ def convert_and_execute_notebook(self, name):
...
 
+    def gen_file_path(self, name):
+        """
+        Returns full path to generated files.  Checks to see if directory
+        exists where generated files are stored and creates one otherwise.
+        """
+        relative_path = self.convert_path(name)
+        file_path = self.get_path("%s.ipynb"%relative_path)
+        parent_path =  rec_apply(os.path.dirname, self.genFileLevel)(file_path)
+        genFileName = name if isinstance(name,str) else name[1] #Name of generated file
```
Comment (rgerkin) [inline sciunit/utils.py]: See above

File: sciunit/utils.py [hunk 5]
```diff
@@ -162,24 +187,43 @@ def convert_and_execute_notebook(self, name):
...
+        genDirPath = self.get_path(os.path.join(parent_path, self.genDirName))
```
Comment (rgerkin) [inline sciunit/utils.py]: See above

**Comment by gsarma** [comment]
Will re-issue after sorting out commit history.

**Pull request closed without merging**